### PR TITLE
Function docs

### PR DESCRIPTION
**Pull request by mattnibs**

### CODE REVIEW

File: docs/language/functions.md [hunk 1]
```diff
@@ -0,0 +1,824 @@
...
+{foo:"aGVsbG8gd29yZA=="}
+```
+
+### `to_hex`
+
+```
+to_hex(b bytes) -> string
+```
+
+`to_hex` encodes byte sequence `b` into a hexadecimal encoded string.
```
Comment (nwt) [inline docs/language/functions.md:152]:
```suggestion
`to_hex` encodes byte sequence `b` into a hexadecimal string.
```

File: docs/language/functions.md [hunk 2]
```diff
@@ -0,0 +1,824 @@
...
+used for arguments that accept a range of value types. They are:
+
+| Name | Types |
+| --- | --- |
+| `<stringy>` | `string`, `bstring`, `err` |
+| `<int>` | `int8`, `int16`, `int32`, `int64`<br />`uint8`, `uint16`, `uint32`, `uint64` |
+| `<float>` | `float32`, `float64` |
+| `<number>` | `<int>`, `<float>` |
+| `<timey>` | `time`, `<number>` |
+| `<any>` | Accepts a value as any type as the argument. |
```
Comment (nwt) [inline docs/language/functions.md:63]: Nit: I think we should sort these.

File: docs/language/functions.md [hunk 3]
```diff
@@ -0,0 +1,824 @@
...
+
+## IPs
+
+### `network_of`
+
+```
+network_of(s ip, [m (net,int,uint)) -> net
+```
+
+`network_of` returns the net of ip address `s` with ip mask `m` applied. `m`
```
Comment (nwt) [inline docs/language/functions.md:173]:
```suggestion
With two arguments, `network_of` returns the net of ip address `s` under mask `m`. `m`
```

File: docs/language/functions.md [hunk 4]
```diff
@@ -0,0 +1,824 @@
...
+can be a net cidr mask or a signed or unsigned integer. If there is no
+optional `m` argument the default mask of `s` is return.
```
Comment (nwt) [inline docs/language/functions.md:175]:
```suggestion
can be a net cidr mask or a signed or unsigned integer. With one argument,
`network_of` returns the default net for `s`.
```

File: docs/language/functions.md [hunk 5]
```diff
@@ -0,0 +1,824 @@
...
+
+### `parse_uri`
+
+```
+parse_uri(u <stringy>) -> record
+```
+
+`parse_uri` parses the [Universal Resource Identifier](https://en.wikipedia.org/wiki/Uniform_Resource_Identifier)
+string `u` into a generic URI record. The returned record has the following
+key/values:
```
Comment (nwt) [inline docs/language/functions.md:337]:
```suggestion
string `u` into a generic URI record. The returned record has the following
type:
```

File: docs/language/functions.md [hunk 6]
```diff
@@ -0,0 +1,824 @@
...
+
+## Value Introspection
+
+### `missing`
+
+```
+missing(e <expression>) -> bool
+```
+
+`missing` returns true if a value in [expression](./expressions.md) e is
```
Comment (nwt) [inline docs/language/functions.md:778]: Nit: We generally don't include a `./` prefix in relative links. (This applies through the file.)
```suggestion
`missing` returns true if a value in [expression](expressions.md) `e` is
```

File: docs/language/functions.md [hunk 7]
```diff
@@ -0,0 +1,824 @@
...
+missing. Typically expression e will be some type of selector expression
```
Comment (nwt) [inline docs/language/functions.md:779]:
```suggestion
missing. Typically `e` is a selector expression 
```

File: docs/language/functions.md [hunk 8]
```diff
@@ -0,0 +1,824 @@
...
+### `missing`
+
+```
+missing(e <expression>) -> bool
+```
+
+`missing` returns true if a value in [expression](./expressions.md) e is
+missing. Typically expression e will be some type of selector expression 
+(e.g. `foo.bar` or `foo[0]`, `foo`) but `missing` will also return true if a
+variable in an generic expression cannot be found (e.g. `foo+1`).
```
Comment (nwt) [inline docs/language/functions.md:781]:
```suggestion
(e.g., `foo.bar` or `foo[0]`, `foo`) but `missing` will also return true if a
variable in a generic expression cannot be found (e.g., `foo+1`).
```

File: docs/language/functions.md [hunk 9]
```diff
@@ -0,0 +1,824 @@
...
+{yes:true,no:false}
+{yes:true,no:false}
+{yes:true,no:false}
+{yes:true,no:false}
+```
+
+### `has`
+
+```
+has(e <expression>) -> bool
```
Comment (nwt) [inline docs/language/functions.md:803]:
```suggestion
has(e ...<expression>) -> bool
```

File: docs/language/functions.md [hunk 10]
```diff
@@ -0,0 +1,824 @@
...
+```
+
+`has` returns true if the value(s) in [expression](./expressions.md) e
+exist. `has` is functionally equivalent to [`!missing(foo)`](#missing).
```
Comment (nwt) [inline docs/language/functions.md:807]:
```suggestion
`has` returns true if a value exists for every [expression](./expressions.md) in the list `e`.
`has` is functionally equivalent to [`!missing(foo)`](#missing).
```
```suggestion
`has` returns true if the value(s) in [expression](./expressions.md) e
exist. `has(e)` is functionally equivalent to [`!missing(e)`](#missing).
```

File: docs/language/functions.md [hunk 11]
```diff
@@ -0,0 +1,824 @@
...
+
+**Output:**
+```mdtest-output
+{id:"1zjJzTWWCJNVrGwqB8kZwhTM2fR"}
+```
+
+### `to_base64`
+
+```
+to_base64(b bytes) -> string
```
[the remark this answers is not in the document]
Comment (mattnibs) [reply docs/language/functions.md:129]: Agreed.

[the remark this answers is not in the document]
Comment (mattnibs) [reply docs/language/functions.md]: https://github.com/brimdata/zed/issues/3255

[the remark this answers is not in the document]
Comment (mattnibs) [reply docs/language/functions.md]: Should this also work with ip addresses?

File: docs/language/functions.md [hunk 12]
```diff
@@ -0,0 +1,823 @@
...
+
+### `unflatten`
+
+```
+unflatten(r record) -> record
+```
+
+`unflatten` returns a copy of r with all successive dotted field names converted
+into nest records. If no argument is supplied to `unflatten`, the root value
+`this` is operated on.
```
Comment (nwt) [inline docs/language/functions.md:478]:
```suggestion
`unflatten` returns a copy of `r` with all dotted field names converted
into nested records. If no argument is supplied to `unflatten`, `unflatten`
operates on `this`.
```

File: docs/language/functions.md [hunk 13]
```diff
@@ -0,0 +1,823 @@
...
+## Strings
+
+### `join`
+
+```
+join(vals [<stringy>], sep <stringy>) -> string
+```
+
+`join` concatenate the elements of string array `vals` to create a single
+string. The string `sep` is placed between each value in the resulting string.
```
Comment (nwt) [inline docs/language/functions.md:500]:
```suggestion
`join` concatenates the elements of string array `vals` to create a single
string. The string `sep` is placed between each value in the resulting string.
```

File: docs/language/functions.md [hunk 14]
```diff
@@ -0,0 +1,823 @@
...
+{foo:["apple","banana","pear","peach"]}
+```
+
+### `to_lower`
+
+```
+to_lower(s <stringy>) -> string
+```
+
+`to_lower` lowercases all unicode letters in `s`.
```
Comment (nwt) [inline docs/language/functions.md:578]:
```suggestion
`to_lower` lowercases all Unicode letters in `s`.
```

File: docs/language/functions.md [hunk 15]
```diff
@@ -0,0 +1,823 @@
...
+{foo:true}
+```
+
+### iserr
+
+```
+iserr(v <any>) -> bool
+```
+
+`iserr` returns true if value v is of type error.
```
Comment (nwt) [inline docs/language/functions.md:706]:
```suggestion
`iserr` returns true if value `v` is of type error.
```

File: docs/language/functions.md [hunk 16]
```diff
@@ -0,0 +1,823 @@
...
+{foo:true}
+```
+
+### nameof
+
+```
+nameof(v <any>) -> string
+```
+
+`nameof` returns the string type name of v if v is an aliased type.
```
Comment (nwt) [inline docs/language/functions.md:725]:
```suggestion
`nameof` returns the string type name of `v` if `v` is an aliased type.
```

File: docs/language/functions.md [hunk 17]
```diff
@@ -0,0 +1,823 @@
...
+
+### `typeof`
+
+```
+typeof(a <any>) -> type
+```
+
+`typeof` returns the [type](../formats/zson.md#357-type-type) of value a.
```
Comment (nwt) [inline docs/language/functions.md:733]:
```suggestion
`typeof` returns the [type](../formats/zson.md#357-type-type) of value `a`.
```

**Review by nwt — APPROVED** [review]

**Pull request merged**